projects/igniteui-angular/src/lib/core/styles/components/chip/_chip-theme.scss

```diff
@@ -0,0 +1,246 @@
+
+        @if (lightness($chip-hover-background) > 50) {
+            $chip-hover-background: lighten($chip-background, 16%);
+            @if not($text-color) and $chip-hover-background{
+                $text-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($order-icon-color) and $chip-hover-background{
+                $order-icon-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($remove-icon-color) and $chip-hover-background{
+                $remove-icon-color: rgba(text-contrast($chip-hover-background), .4);
+            }
+
+            @if not($remove-icon-hover-color) and $chip-hover-background{
+                $remove-icon-hover-color: rgba(text-contrast($chip-hover-background), 1);
+            }
+        } @else {
+            $chip-hover-background: darken($chip-background, 12%);
+
+            @if not($text-color) and $chip-hover-background{
+                $text-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($order-icon-color) and $chip-hover-background{
+                $order-icon-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($remove-icon-color) and $chip-hover-background{
+                $remove-icon-color: rgba(text-contrast($chip-hover-background), .4);
+            }
+
+            @if not($remove-icon-hover-color) and $chip-hover-background{
+                $remove-icon-hover-color: rgba(text-contrast($chip-hover-background), 1);
+            }
+        }
+    }
+
+    @if not($text-hover-color) and $chip-hover-background{
+        $text-hover-color: text-contrast($chip-hover-background);
+    }
+
+    @if not($remove-icon-hover-color) and $chip-hover-background{
+        $remove-icon-hover-color: rgba(text-contrast($chip-hover-background), .4);
+    }
+
+    @if not($remove-icon-color) and $chip-hover-background{
+        $remove-icon-color: rgba(text-contrast($chip-hover-background), .4);
+    }
+
+    @if not($order-icon-hover-color) and $chip-hover-background{
+        $order-icon-hover-color: text-contrast($chip-hover-background);
+    }
+
+
+    @return extend($default-theme, (
+        palette: $palette,
+        chip-background: $chip-background,
+        chip-hover-background: $chip-hover-background,
+
+        text-color: $text-color,
+        text-hover-color: $text-hover-color,
+        order-icon-hover-color: $order-icon-hover-color,
+
+        order-icon-color: $order-icon-color,
+        dir-icon-color: $dir-icon-color,
+        remove-icon-color: $remove-icon-color,
+        remove-icon-hover-color: $remove-icon-hover-color,
+    ));
+}
+
+/// @param {Map} $theme - The theme used to style the component.
+/// @requires {mixin} igx-root-css-vars
+/// @requires rem
+/// @requires var
+@mixin igx-chip($theme) {
+    @include igx-root-css-vars($theme);
+
+    $transition: all 120ms $ease-in-out-cubic;
+
+    %igx-chips-area {
```

Comment:
As before, I don't understand why this is part of a theme meant to style a single chip instance.